extensions/parquet/base/src/main/java/io/deephaven/parquet/base/ColumnChunkReader.java

```diff
@@ -32,9 +34,24 @@ public interface ColumnChunkReader {
      */
     int getMaxRl();
 
+    /**
+     * @return The offset index for this column chunk, or null if it not found in the metadata.
+     */
+    @Nullable
+    OffsetIndex getOffsetIndex();
+
     interface ColumnPageReaderIterator extends Iterator<ColumnPageReader>, AutoCloseable {
         @Override
         void close() throws IOException;
+
+        /**
+         * Directly access a page reader for a given page number. This is an optional method that may not be
+         * implemented. Note that the user should either use {@link Iterator} methods or this method, but not both.
+         */
+        @Nullable
+        default ColumnPageReader getPageReader(final int pageNum) {
```

Comment:
IMO, we should add a method `boolean supportsDirectAccess()` and make sure our implementations have proper assertions (for example, in `OffsetIndexBasedColumnChunkPageStore`)

Reply:
The other half of me thinks this should be a _separate_ interface, not merged w/ `ColumnPageReaderIterator`. Are there any cases where a consumer doesn't know their exact access patterns a priori?

```java
...
    interface ColumnPageDirectorAccessor extends AutoCloseable {
        ColumnPageReader getPageReader(final int pageNum);
        
        @Override
        void close() throws IOException;
    }
    
    ColumnPageReaderIterator getPageIterator() throws IOException;
    
    ColumnPageDirectorAccessor getPageAccessor() throws IOException;
    ...
```